frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -166,12 +179,82 @@ public void after(java.util.List<? extends VFileEvent> events) {
 
   private void subscribeToGitRepositoryFilesChanges() {
     Topic<GitRepositoryChangeListener> topic = GitRepository.GIT_REPO_CHANGE;
-    GitRepositoryChangeListener listener = repository -> queueRepositoryUpdateAndModelRefresh();
+    GitRepositoryChangeListener listener = repository -> {
+      trackCurrentBranchChange(repository);
+      queueRepositoryUpdateAndModelRefresh();
+    };
+
     val messageBusConnection = project.getMessageBus().connect();
     messageBusConnection.subscribe(topic, listener);
     Disposer.register(project, messageBusConnection);
   }
 
+  @IgnoreUIThreadUnsafeCalls
+  private void notifyAboutUnmanagedBranch(String branchName) {
+    val gitRepositorySelectionProvider = getGitRepositorySelectionProvider();
+    val gitRepository = gitRepositorySelectionProvider.getSelectedGitRepository();
+    if (gitRepository == null) {
+      LOG.warn("Selected repository is null");
+      return;
+    }
+
+    Path macheteFilePath = gitRepository.getMacheteFilePath();
+    boolean isMacheteFilePresent = Files.isRegularFile(macheteFilePath);
+
+    if (!isMacheteFilePresent) {
+      LOG.warn("Machete file (${macheteFilePath}) is absent, so no unmanaged branch notification will show up");
+      return;
+    }
+
+    val repository = gitMacheteRepository;
+    if (repository == null) {
+      LOG.warn("gitMacheteRepository is null, so no unmanaged branch notification will show up");
+      return;
+    }
+
+    if (gitMacheteRepositorySnapshot == null) {
+      LOG.warn("gitMacheteRepositorySnapshot is null, so no unmanaged branch notification will show up");
+      return;
+    }
+
+    val eligibleLocalBranchNames = gitMacheteRepositorySnapshot.getManagedBranches().map(IManagedBranchSnapshot::getName)
+        .toSet();
+
+    new SlideInUnmanagedBranch(
+        project,
+        () -> Try.of(() -> repository.inferParentForLocalBranch(eligibleLocalBranchNames, branchName)),
+        getGitRepositorySelectionProvider(),
+        inferredParent -> ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL, () -> {
+          val showForThisProject = UnmanagedBranchNotificationFactory.showForThisProject(project);
```

Comment:
Maybe `shouldShowFor...`, coz the current names suggest that the method show something rather than just return a boolean whether to show or not